libs/ai-endpoints/langchain_nvidia_ai_endpoints/tools.py

```diff
@@ -0,0 +1,272 @@
+"""OpenAI chat wrapper."""
+
+from __future__ import annotations
+
+import logging
+from operator import itemgetter
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Sequence,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from langchain_core.language_models import LanguageModelInput
+from langchain_core.messages import BaseMessage
+from langchain_core.output_parsers.base import OutputParserLike
+from langchain_core.output_parsers.openai_tools import (
+    JsonOutputKeyToolsParser,
+    PydanticToolsParser,
+)
+from langchain_core.pydantic_v1 import BaseModel
+from langchain_core.runnables import Runnable, RunnableMap, RunnablePassthrough
+from langchain_core.tools import BaseTool
+from langchain_core.utils.function_calling import convert_to_openai_tool
+
+logger = logging.getLogger(__name__)
+
+_BM = TypeVar("_BM", bound=BaseModel)
+
+
+## Directly Inspired by OpenAI/MistralAI's server-side support.
+## Moved here for versioning/additional integration options.
+
+
+"""
```

Comment:
what is this code string for? Causes linting to fail because it has some `from langchain.x import y` imports that aren't recommended in the package. Easiest way to fix CI will be to just remove this string!

Reply:
it's experimental code, which could be removed for now. the bind_tools() does not need to be exposed yet either.